### PR TITLE
Print a better error message if we are unable to read / write auth token from / to the cache

### PR DESCRIPTION
**Pull request by Kami**
The title says it all.

### CODE REVIEW

**Comment by Kami** [comment]
/cc @dzimine 

Better?

File: st2client/st2client/shell.py [hunk 1]
```diff
@@ -416,14 +416,19 @@ def _get_cached_auth_token(self, client, username, password):
 
         if not os.access(ST2_CONFIG_DIRECTORY, os.R_OK):
             # We don't have read access to the file with a cached token
-            LOG.warn('User "%s" doesn\'t have read access to file "%s"' % (os.getlogin(),
-                                                                           CACHED_TOKEN_PATH))
+            message = ('Unable to retrieve cached token from "%s" (user %s doesn\'t have read '
+                       'access to the parent directory). Subsequent requests won\'t use a '
+                       'cached token meaning they will be slower.' % (CACHED_TOKEN_PATH,
```
Comment (jfryman) [inline st2client/st2client/shell.py]: "and may be slower". I mean, it will be because of the round trip, but, we're not talking huge amounts here.

Comment (Kami) [inline st2client/st2client/shell.py]: yep, will make the change :)

**Comment by lakshmi-kannan** [comment]
L.G.T.M

**Pull request merged**